iModelJsNodeAddon/JsCloudSqlite.cpp

```diff
@@ -384,6 +384,53 @@ struct JsCloudContainer : CloudContainer, Napi::ObjectWrap<JsCloudContainer> {
         return value;
     }
 
+    Napi::Value QueryBcvStats(NapiInfoCR info) {
+        RequireConnected();
+        bool addClientInformation = false;
+        if (info[0].IsObject()) {
+            auto filterOpts = BeJsConst(info[0]);
+            addClientInformation = filterOpts[JSON_NAME(addClientInformation)].asBool();
+        }
+
+        Statement stmt;
+        Utf8CP statNames [3] = { "nlock", "ncache", "cachesize" };
+        Utf8CP jsNames [3] = { "lockedCacheslots", "populatedCacheslots", "totalCacheslots" };
+        auto rc = stmt.Prepare(m_containerDb, "SELECT value FROM bcv_stat where name = ?");
+        BeAssert (rc == BE_SQLITE_OK);
+        UNUSED_VARIABLE(rc);
+        BeJsNapiObject value(Env());
+        for (int i = 0; i < sizeof(statNames) / sizeof(statNames[0]); i++) {
+            stmt.BindText(1, statNames[i], Statement::MakeCopy::No);
+            auto result = stmt.Step();
+            if (result != BE_SQLITE_ROW) {
+                value[jsNames[i]] = -1; 
+            } else {
+                value[jsNames[i]] = stmt.GetValueInt(0); 
+            }
+            stmt.Reset();
+            stmt.ClearBindings();
+        }
+        stmt.Finalize();
+        // TODO: Would a where clause speed up the performance of this query?

```

Comment:
@khanaffan do you have an opinion on this TODO? 
The bcv_database virtual table will have one row for every database in an attached container, which seems like it could grow to a decent size, since there is one database for each checkpoint. I could add a where clause like 'where nclient >= 1' to only grab databases which have some open connection to them, but not sure if its worth doing.

Reply:
I would not worry about it. Unless you get to a million files per container which seem unrealistic.